File: puz1.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"slices"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanLines)
	var l, r int
	ll := make([]int, 0)
	lr := make([]int, 0)

	for scanner.Scan() {
		line := scanner.Text()
		_, err := fmt.Sscanf(line, "%d %d", &l, &r)
		if err != nil {
			panic("failed to parse line")
		}
		ll = append(ll, l)
		lr = append(lr, r)
	}

	slices.Sort(ll)
	slices.Sort(lr)

	counter := make(map[int]int)

	for _, v := range lr {
		counter[v] = counter[v] + 1
	}

	diff := 0
	sim_score := 0

	for idx, v := range ll {
		rv := lr[idx]
		abs := int(math.Abs(float64(rv - v)))
		diff = diff + abs
		sim_score = sim_score + v*counter[v]
	}

	fmt.Println(diff)
	fmt.Println(sim_score)
}

File: puz1/puz1.py
import sys
from collections import Counter

l_list = []
r_list = []
for line in sys.stdin:
    l, r = line.split()
    l_list.append(int(l))
    r_list.append(int(r))

r_counter = Counter(r_list)
l_list.sort()
r_list.sort()

diff = sum(abs(y -x) for x, y in zip(l_list, r_list))
sim_score = sum(x * r_counter.get(x, 0) for x in l_list)

print(diff)
print(sim_score)
